[Translation] check empty notes

davidvancl opened the pull request:
| Q             | A
| ------------- | ---
| Branch?       | 6.4
| Bug fix?      | yes
| New feature?  | no 
| Deprecations? | no
| Issues        | --
| License       | MIT

In the case of notes, an empty array is always stored in the metadata. The is_iterable condition always returns true even if notes is empty and there is no location info. When notes are loaded within a translation, translations containing <notes/> will crash.

davidvancl committed:
[Translation] check empty notes

OskarStark commented:
Does this also apply to 6.4?

Please add a testcase

welcoMattic commented on src/Symfony/Component/Translation/Dumper/XliffFileDumper.php:
```diff
@@ -176,7 +176,7 @@ private function dumpXliff2(string $defaultLocale, MessageCatalogue $messages, ?
 
             // Add notes section
-            if ($this->hasMetadataArrayInfo('notes', $metadata)) {
+            if ($this->hasMetadataArrayInfo('notes', $metadata) && !empty($metadata['notes'] ?? null)) {
```
```suggestion
            if ($this->hasMetadataArrayInfo('notes', $metadata) && $metadata['notes'] !== []) {
```

We avoid `empty` usage, because its behavior too implicit.

Could you add a test case to cover this fix please?

welcoMattic commented:
Thank you @davidvancl. 
Could you take care of what fabbot reported? And be sure that failing tests are not related to your modifications?

davidvancl commented:
I added a test for empty notes, I don't know why all the tests failed. When I run them locally they pass. Could it be somehow related to CI? also fixed the syntax that was reporting fabbot. @welcoMattic Thank you.

nicolas-grekas commented on src/Symfony/Component/Translation/Dumper/XliffFileDumper.php:
```diff
@@ -176,7 +176,7 @@ private function dumpXliff2(string $defaultLocale, MessageCatalogue $messages, ?
 
             // Add notes section
-            if ($this->hasMetadataArrayInfo('notes', $metadata)) {
+            if ($this->hasMetadataArrayInfo('notes', $metadata) && [] !== $metadata['notes']) {
```
we don't do empty-value comparisons:
```suggestion
            if ($this->hasMetadataArrayInfo('notes', $metadata) && $metadata['notes']) {
```

nicolas-grekas commented:
Thank you @davidvancl.

Hanmac commented:
@nicolas-grekas now my tests are failing:

```
-   <unit id="Ll7LrI6" name="empty">\n
+    <unit id="Lhz6grA" name="empty">\n
```
